src/bot/commands/ignore.ts

```diff
@@ -0,0 +1,193 @@
+        const isAlreadyIgnored = toggledCategories.includes(toggledCategory);
+
+        toggledCategories = isAlreadyIgnored
+          ? toggledCategories.filter((category) => category !== toggledCategory)
+          : [...toggledCategories, toggledCategory];
+
+        await Promise.all([
+          interaction.editReply(this.generateMessageOptions(toggledCategories)),
+          buttonInteraction.deferUpdate()
+        ]);
+
+        continue;
+      }
+
+      if (buttonInteraction.customId === 'discard') {
+        await Promise.all([
+          interaction.editReply(interactionError("Vos modifications n'ont pas été enregistrées.")),
+          buttonInteraction.deferUpdate()
+        ]);
+        toggledCategories = godfather.ignored_categories;
+
+        break;
+      }
+
+      if (buttonInteraction.customId === 'save') {
+        await Promise.all([
+          interaction.editReply(interactionValidate('Vos modifications ont bien été enregistrées.')),
+          buttonInteraction.deferUpdate()
+        ]);
+
+        break;
+      }
+    }
+    toggledCategories.sort();
+    godfather.ignored_categories.sort();
+
+    if (
+      toggledCategories.length === godfather.ignored_categories.length &&
+      toggledCategories.every((category, index) => category === godfather.ignored_categories[index])
+    ) {
+      // Categories are the same
+      return;
+    }
+
+    await prisma.godfather.update({
+      where: {
+        id: godfather.id
+      },
+      data: {
+        ignored_categories: toggledCategories
+      }
+    });
+  }
+
+  private generateMessageOptions(ignoredCategories: Category[]): InteractionReplyOptions {
+    const embed = info('Choisissez quels catégories vous souhaitez ignorer / voir à nouveau');
+    const buttons = Object.entries(CategoriesRefs).map(([category, name]) => {
+      const isAlreadyIgnored = ignoredCategories.includes(category as Category);
+
+      return {
+        type: ComponentType.Button,
+        custom_id: category,
+        style: isAlreadyIgnored ? ButtonStyle.Secondary : ButtonStyle.Primary,
+        label: isAlreadyIgnored ? `❌ ${name}` : `✅ ${name}`
+      };
+    });
+    const buttonsRows = [buttons.slice(0, 3), buttons.slice(3)];
+
+    return {
+      embeds: [embed],
+      components: [
+        ...buttonsRows.map((row) => ({
+          type: ComponentType.ActionRow,
+          components: row
+        })),
+        {
+          type: ComponentType.ActionRow,
+          components: [
+            {
+              type: ComponentType.Button,
+              custom_id: 'discard',
+              label: IgnoreCommand.ActionButtonsMapping.discard,
+              style: ButtonStyle.Danger
+            },
+            {
+              type: ComponentType.Button,
+              custom_id: 'save',
+              label: IgnoreCommand.ActionButtonsMapping.save,
+              style: ButtonStyle.Success
+            }
+          ]
+        }
+      ]
+    } as InteractionReplyOptions;
```

Comment:
Pas de raison de forcer le type
```suggestion
    };
```